_studio/mfx_lib/encode_hw/vp9/include/mfx_vp9_encode_hw_utils.h

```diff
@@ -965,11 +960,11 @@ inline void CopySegmentationBuffer(mfxExtVP9Segmentation & dst, mfxExtVP9Segment
 {
     mfxU8* tmp = dst.SegmentId;
     ZeroExtBuffer(dst);
-    memcpy_s(&dst, sizeof(mfxExtVP9Segmentation), &src, sizeof(mfxExtVP9Segmentation));
+    dst = src;
     dst.SegmentId = tmp;
     if (dst.SegmentId && src.SegmentId && dst.NumSegmentIdAlloc)
     {
-        memcpy_s(dst.SegmentId, dst.NumSegmentIdAlloc, src.SegmentId, dst.NumSegmentIdAlloc);
+    	dst.SegmentId == src.SegmentId;
```

Comment:
first of all, operator == definitely not what you wanted. Second - this is an array, so you have to perform fair deep copy of it

Reply:
oh!!! stupid typo!